Sources/VaporAWSLambdaRuntime/APIGatewayV2.swift

```diff
@@ -82,34 +91,70 @@ extension APIGateway.V2.Request: Vapor.StorageKey {
+        // Create the promise
+        let promise = context.eventLoop.makePromise(of: APIGateway.V2.Response.self)
+
+        // Create the headers
+        var headers = [String: String]()
         response.headers.forEach { name, value in
-            var values = headers[name] ?? [String]()
-            values.append(value)
-            headers[name] = values
+            if let current = headers[name] {
+                headers[name] = "\(current),\(value)"
+            } else {
+                headers[name] = value
+            }
         }
 
+        // Can we access the body right away?
         if let string = response.body.string {
-            self = .init(
+            promise.succeed(.init(
                 statusCode: AWSLambdaEvents.HTTPResponseStatus(code: response.status.code),
-                multiValueHeaders: headers,
+                headers: headers,
                 body: string,
                 isBase64Encoded: false
-            )
-        } else if var buffer = response.body.buffer {
-            let bytes = buffer.readBytes(length: buffer.readableBytes)!
-            self = .init(
+            ))
+        } else if let bytes = response.body.data {
+            promise.succeed(.init(
                 statusCode: AWSLambdaEvents.HTTPResponseStatus(code: response.status.code),
-                multiValueHeaders: headers,
+                headers: headers,
                 body: String(base64Encoding: bytes),
                 isBase64Encoded: true
-            )
+            ))
         } else {
-            self = .init(
-                statusCode: AWSLambdaEvents.HTTPResponseStatus(code: response.status.code),
-                multiValueHeaders: headers
-            )
+            // See if it is a stream and try to gather the data
+            response.body.collect(on: context.eventLoop).whenComplete { collectResult in
+
+                switch collectResult {
+                case let .failure(error):
+                    promise.fail(error)
+
+                case let .success(buffer):
+
+                    // Was there any content
+                    guard
+                        var buffer = buffer,
+                        let bytes = buffer.readBytes(length: buffer.readableBytes)
+                    else {
+                        promise.succeed(.init(
+                            statusCode: AWSLambdaEvents.HTTPResponseStatus(code: response.status.code),
+                            headers: headers
+                        ))
+
+                        return
+                    }
+
+                    // Done
+                    promise.succeed(.init(
+                        statusCode: AWSLambdaEvents.HTTPResponseStatus(code: response.status.code),
+                        headers: headers,
+                        body: String(base64Encoding: bytes),
+                        isBase64Encoded: true
+                    ))
+                }
```

Comment:
I think we can make the code here a lot simpler. @fcobia wdyt?

```suggestion
            // See if it is a stream and try to gather the data
            return response.body.collect(on: context.eventLoop).map { (buffer) -> APIGateway.V2.Response in
                // Was there any content
                guard
                    var buffer = buffer,
                    let bytes = buffer.readBytes(length: buffer.readableBytes)
                else {
                    return APIGateway.V2.Response(
                        statusCode: AWSLambdaEvents.HTTPResponseStatus(code: response.status.code),
                        headers: headers
                    )
                }

                // Done
                return APIGateway.V2.Response(
                    statusCode: AWSLambdaEvents.HTTPResponseStatus(code: response.status.code),
                    headers: headers,
                    body: String(base64Encoding: bytes),
                    isBase64Encoded: true
                )
            }
```